src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -394,6 +406,12 @@ export const WalletProvider = ({ children }: { children: React.ReactNode }): JSX
               if (localWalletType === 'tallyho' && window?.ethereum?.isMetaMask) disconnect()
               const localTallyHoWallet = await state.adapters.get(KeyManager.TallyHo)?.pairDevice()
               if (localTallyHoWallet) {
+                const chainId = await localTallyHoWallet.ethGetChainId?.()
+                if (bnOrZero(chainId).toString() !== CHAIN_REFERENCE.EthereumMainnet) {
+                  await localTallyHoWallet.ethSwitchChain?.(
+                    bn(CHAIN_REFERENCE.EthereumMainnet).toNumber(),
+                  )
```

Comment:
Note: This is currently a no-op. I managed to get to a chainId other than 1, and thus to this block, by building Tally from source and enabling all the multichain dev flags:
```
SUPPORT_POLYGON=true
SUPPORT_ARBITRUM=true
SUPPORT_OPTIMISM=true
```
However, `wallet_switchEthereumChain`, while broadcasted, isn't actually picked by Tally yet.
Until this gets released in Tally, this is a dead block, as the condition will always evaluate to false. IMO, it's better to still keep it here for consistency.

Reply:
Same here, can we add a comment in the code?

Reply:
Added in bdb09a0bad87b7e7ffe3ce0a5e0fbdf2bfdf4018 !